src/syscalls/native.rs

```diff
@@ -581,3 +581,66 @@ pub fn exec(
         )
     }
 }
+
+#[repr(C)]
+pub struct SpawnArgs {
+    pub memory_limit: u64,
+    pub exit_code: *mut i8,
+    pub content: *mut u8,
+    pub content_length: *mut u64,
+}
+
+/// The Spawn and the latter two syscalls: Get Memory Limit and Set Content
+/// together, implement a way to call another CKB Script in a CKB Script.
+/// Note: available after ckb2023.
+///
+/// Returns success or a syscall error.
+pub fn spawn(index: usize, source: Source, bounds: usize, argv: &[&CStr], spgs: &SpawnArgs) -> u64 {
+    let argc = argv.len();
+    let mut argv_ptr = alloc::vec![core::ptr::null(); argc + 1];
+    for (idx, cstr) in argv.into_iter().enumerate() {
+        argv_ptr[idx] = cstr.as_ptr();
+    }
+    unsafe {
+        syscall(
+            index as u64,
+            source as u64,
+            bounds as u64,
+            argc as u64,
+            argv_ptr.as_ptr() as u64,
+            spgs as *const SpawnArgs as u64,
+            0,
+            SYS_SPAWN,
+        )
+    }
+}
+
+/// Get memory limit.
+/// Note: available after ckb2023.
+///
+/// Returns a number between 1 and 8, representing 0.5 to 4M of memory.
+pub fn get_memory_limit() -> u64 {
+    unsafe { syscall(0, 0, 0, 0, 0, 0, 0, SYS_GET_MEMORY_LIMIT) }
+}
+
+/// Set content.
+/// Note: available after ckb2023.
+///
+/// Return the actual written data length or a syscall error.
```

Comment:
Add more comments: when content length given at `spawn` is smaller than size of `buf` here, it returns smaller one.